meilisearch-core/src/error.rs

```diff
@@ -176,6 +176,16 @@ pub enum FacetError {
     InvalidDocumentAttribute(String),
 }
 
+impl FacetError {
+    pub fn unexpected_token(expected: &'static [&'static str], found: String) -> FacetError {
+        FacetError::UnexpectedToken{ expected, found }
+    }
+
+    pub fn attribute_not_set(expected: Vec<String>, found: String) -> FacetError {
+        FacetError::AttributeNotSet{ expected, found }
+    }
```

Comment:
We can reduce the line length of the other file by avoiding calling `to_string` and transform it there.

```suggestion
    pub fn unexpected_token(expected: &'static [&'static str], found: impl ToString) -> FacetError {
        FacetError::UnexpectedToken{ expected, found: found.to_string() }
    }

    pub fn attribute_not_set(expected: Vec<String>, found: impl ToString) -> FacetError {
        FacetError::AttributeNotSet{ expected, foundL found.to_string() }
    }
```